streams/src/main/java/org/apache/kafka/streams/kstream/internals/foreignkeyjoin/ForeignJoinSubscriptionSendProcessorSupplier.java

```diff
@@ -83,9 +84,15 @@ public void init(final ProcessorContext context) {
             valueSerdeTopic = valueSerdeTopicSupplier.get();
             // get default key serde if it wasn't supplied directly at construction
             if (foreignKeySerializer == null) {
+                if (context.keySerde() == null || context.keySerde().serializer() == null) {
```

Comment:
I don't think that a `Serde` should ever wrap `null` ? Can we omit the second check?